internal/gps/identifier.go

```diff
@@ -40,6 +40,11 @@ import (
 // the type system when a path-ish string must have particular semantics.
 type ProjectRoot string
 
+// ProjectRootPtr takes a ProjectRoot value and returns a pointer.
+func ProjectRootPtr(p ProjectRoot) *ProjectRoot {
```

Comment:
I think we should remove this and make it an unexported part of the tests. Since it's not used outside the tests I think this is not providing value to any part of the implementation.

I'll move it to the test files.